### PR TITLE
Write walker configuration.

### PR DESCRIPTION
**Pull request by gbalduzz**
This PR stores the configuration cross DCA iterations for each walker, and optionally dump it to disk at the end (could be modified to write at each iteration to restart from a crash).
This is implemented only for the CT-AUX solver. The configuration is stored as a buffer of raw bytes where the minimal information necessary to recover each vertex is stored, as the vertex class stores a bunch of metadata and pointers.
This could have been done as well with a subclass of the vertex, but I believe this generalizes better to different solvers and configurations, as only the read and write to buffer operations needs to be defined.
This will need a non trivial merge with #61

### CODE REVIEW

**Comment by PDoakORNL** [comment]
What would you expect the memory footprint of this to be? Per walker

**Comment by gbalduzz** [comment]
Sorry for the delay.
It is (8 + 44 * expansion_order) bytes. So for a typical 1k expansion order, we have approximately  50 kB per walker.

**Comment by PDoakORNL** [comment]
Can you resolve the conflicts @gbalduzz.

**Comment by gbalduzz** [comment]
Sorry, I pushed to the wrong branch `write_walker_configuration`. I pushed the merge to the correct one now.

**Comment by gbalduzz** [comment]
Note for the Jenkins test: Daint is down for maintenance until tomorrow eve.

**Comment by PDoakORNL** [comment]
I will run this through my CI this evening.

**Comment by PDoakORNL** [comment]
retest this please

**Comment by PDoakORNL** [comment]
Passes GPU and CPU CI on CONDO

File: include/dca/math/util/vector_operations.hpp [hunk 1]
```diff
@@ -245,6 +245,17 @@ std::enable_if_t<std::is_arithmetic<T>::value, bool> isSameVector(const std::vec
   return (distance2(x, y) < tol);
 }
 
+// Compare two vectors element-wise.
+template <typename T>
+bool operator==(const std::vector<T>& x, const std::vector<T>& y) {
+  if (x.size() != y.size())
+    return false;
+  for (int i = 0; i < x.size(); ++i)
+    if (!(x[i] == y[i]))
```
[the remark this answers is not in the document]
Comment (gbalduzz) [reply include/dca/math/util/vector_operations.hpp]: There is not a != operator for one of the types it is used on. I could add it, or use !(==) as done here.

**Pull request merged**